etl/steps/data/garden/regions/2023-01-01/migration.py

```diff
@@ -0,0 +1,884 @@
+    # New column "region_type" will help distinguish between countries, continents, sub-country regions, or other special
+    # kinds of regions.
+    # By default, assume all regions are countries (since that would be the majority of cases).
+    df["region_type"] = "country"
+    # New column "is_historical" will be True for those regions that do not exist today.
+    # By default, assume all rows correspond to non-historical regions.
+    df["is_historical"] = False
+    # New column "defined_by" will help distinguish between regions defined by OWID, and by other institutions.
+    df["defined_by"] = "owid"
+    # New column "end_year" will not be empty for historical regions, and will inform of the last year of the region.
+    df["end_year"] = pd.NA
+    # New column "successors" will not be empty for historical regions, and will give the list of regions that occupied
+    # the same geographical land after the region stoppted existing.
+    df["successors"] = pd.NA
+    # New column "related" will not be empty if there are related member regions that, for some datasets, could lead
+    # to double-counting. For example, "China" has the related member "Hong-Kong", since some data providers include
+    # Hong-Kong's data in China, and also provide data for Hong-Kong.
+    df["related"] = pd.NA
+    # Empty the existing column for members.
+    df["members"] = pd.NA
+
+    # Apply some minor corrections to existing data.
+
+    # Add a short name for DRC.
+    df.loc[df["code"] == "COD", "short_name"] = "DR Congo"
+
+    # Assign TLS the right name and short name, which is East Timor.
+    df.loc[df["code"] == "TLS", "name"] = "East Timor"
+    df.loc[df["code"] == "TLS", "short_name"] = "East Timor"
+
+    # Add a short name for DRC.
+    df.loc[df["code"] == "COD", "short_name"] = "DR Congo"
+
+    # Assign TLS the right name and short name, which is East Timor.
+    df.loc[df["code"] == "TLS", "name"] = "East Timor"
+    df.loc[df["code"] == "TLS", "short_name"] = "East Timor"
+
+    # Edit region type of continents.
+    for continent in continents:
+        df.loc[df["name"] == continent, "region_type"] = "continent"
+
+    # Edit region type of special regions.
+    for region in aggregates:
+        df.loc[df["name"] == region, "region_type"] = "aggregate"
+
+    # Edit region type of other regions.
+    for region in other_regions:
+        df.loc[df["name"] == region, "region_type"] = "other"
+
+    # Remove unnecessary regions.
+    df = df[~df["name"].isin(regions_to_remove)].reset_index(drop=True)
+
+    # Edit the rows corresponding to historical regions.
+    for historical_region in historical_regions:
+        end_year = historical_regions[historical_region]["end_year"]
+        successors = historical_regions[historical_region]["successors"]
+        successor_codes = sorted([df[df["name"] == successor]["code"].item() for successor in successors])
+        region_sel = df["name"] == historical_region
```

Comment:
We could maybe assert that `sum(region_se) == 1`.

Reply:
I just added a similar check above, ensuring that there are no repeated names.